components/playlist/README.md

```diff
@@ -0,0 +1,100 @@
+# Playlists API
+
+## Calling the API from C++
+
+You will need some or all of the following headers:
+
+``` c++
+#include "brave/browser/playlists/playlists_service_factory.h"
+#include "brave/components/playlists/browser/playlists_controller.h"
+#include "brave/components/playlists/browser/playlists_controller_observer.h"
+#include "brave/components/playlists/browser/playlists_media_file_controller.h"
+#include "brave/components/playlists/browser/playlists_service.h"
+#include "brave/components/playlists/browser/playlists_types.h"
+```
+
+The Playlists service is a `BrowserContextKeyedService`, so you will access it through the `PlaylistsServiceFactory`.
+
+``` c++
+PlaylistsService* playlists_service = PlaylistsServiceFactory::GetInstance()->
+  GetForProfile(SOME_PROFILE);
+```
+
+There are only two interesting things in the `PlaylistsService`: an `Init()` method and a `controller()` method. You need to call the `Init()` method exactly once so it can set up the internal database it uses to track local playlists. It takes no parameters and returns immediately, but the Playlist API is not immediately ready. You can use the controller to check whether initialization is complete.
```

Comment:
I'm pretty sure we should keep all this implementation descriptions in the corresponding header files